### PR TITLE
Clean up dead code (#36138)

### LINKED ISSUE #36138: Clean up dead code in AbstractLegacyProject_IIntellisenseBuildTarget.cs
https://github.com/dotnet/roslyn/blob/master/src/VisualStudio/Core/Def/Implementation/ProjectSystem/Legacy/AbstractLegacyProject_IIntellisenseBuildTarget.cs#L15-L72

### PR DESCRIPTION
**Pull request by nbalakin**
Fix #36138

### CODE REVIEW

File: src/VisualStudio/Core/Def/Implementation/ProjectSystem/Legacy/AbstractLegacyProject_IIntellisenseBuildTarget.cs [hunk 1]
```diff
@@ -1,77 +1,13 @@
...
-using System.IO;
-using Microsoft.CodeAnalysis;
-using Microsoft.CodeAnalysis.Diagnostics;
-using Microsoft.CodeAnalysis.Internal.Log;
 using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.Interop;
-using Roslyn.Utilities;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.Legacy
 {
     internal partial class AbstractLegacyProject : IIntellisenseBuildTarget
```
Comment (nbalakin) [inline src/VisualStudio/Core/Def/Implementation/ProjectSystem/Legacy/AbstractLegacyProject_IIntellisenseBuildTarget.cs]: Is it safe to remove the whole `IIntellisenseBuildTarget` realization?

**Comment by jasonmalinowski** [comment]
So I agree this code is dead in that "it's not doing anything _today_" but I'm not sure we should be deleting it entirely versus fixing it up. This is code I commented out as a part of #26931, because it was necessary while doing that refactoring to comment out some bits to keep other code building so I could get the core working. That _should_ have been fixed up -- that it was missed was an outright mistake.

So now that said, @tmeschter you do still call into this, right? And if design time builds fail do we have any better way we show that now?

**Comment by tmeschter** [comment]
@jasonmalinowski I think we're still calling this. I can't think of another way to surface this off the top of my head.

**Comment by jinujoseph** [comment]
cc @tmat

**Comment by jasonmalinowski** [comment]
Apologies @nbalakin this one shouldn't have been marked as help wanted, or at least it's not trivial. We do need to restore that code that is commented out and make it work again. The high level design is if that's given an error we raise the error in the error list. I'm not sure how much work it will take it get working again without digging into it myself. You're welcome to give it a shot but I can't exactly predict easy success.

**Pull request closed without merging**